### PR TITLE
CBO-639: Item numbering

### PR DESCRIPTION
**Pull request by cleargif**
#### What
Defendants: 
- moved the remove link
- refactor to the view for non-sidebar pages

General numbering:
- Numbers appear for +2 items 
- Remove links appear for +2 items
- Better scoping for event binding in the jQuery plugin

#### Ticket

[Numbering for single item pages](https://dsdmoj.atlassian.net/browse/CBO-639)

### CODE REVIEW

File: app/views/external_users/claims/defendants/_defendant_fields.html.haml [hunk 1]
```diff
@@ -30,13 +32,10 @@
             %p
               = t('.order_for_judicial_apportionment_help')
 
-    .documents
+    .documents.fx-numberedList-item_
```
Comment (cleargif) [inline app/views/external_users/claims/defendants/_defendant_fields.html.haml:35]: The _ is intentional

File: app/views/external_users/claims/defendants/_representation_order_fields.html.haml [hunk 2]
```diff
@@ -3,7 +3,8 @@
   %a{id: "defendant_#{@defendant_count+1}_representation_order_#{@representation_order_count}_representation_order_date"}
   %h3.heading-medium
     = t('.representation_order')
-
+    %span.fx-numberedList-number_
+    = link_to_remove_association t('.remove'), f, class: "header-action fx-numberedList-action_"
```
Comment (cleargif) [inline app/views/external_users/claims/defendants/_representation_order_fields.html.haml:7]: The _ in intentional

File: app/views/external_users/claims/misc_fees/litigators/_misc_fee_fields.html.haml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     %span#fx-numberedList-heading
       = t('.misc_fee')
       %span.fx-numberedList-number
-    = link_to_remove_association t('.remove'), f, { wrapper_class: 'misc-fee-group', class: "header-action" }
+    = link_to_remove_association t('.remove'), f, wrapper_class: 'misc-fee-group', class: "header-action fx-numberedList-action", style: "display: none;"
```
Comment (cleargif) [inline app/views/external_users/claims/misc_fees/litigators/_misc_fee_fields.html.haml:6]: Using the inline style as the jquery will simply update it inline as well

File: app/views/external_users/claims/misc_fees/advocates/_misc_fee_fields.html.haml [hunk 4]
```diff
@@ -3,7 +3,7 @@
   %h2.bold-medium
     = t('.misc_fee')
     %span.fx-numberedList-number
-    = link_to_remove_association t('.remove'), f, { wrapper_class: 'misc-fee-group', class: "header-action" }
+    = link_to_remove_association t('.remove'), f, wrapper_class: 'misc-fee-group', class: "header-action fx-numberedList-action", style: "display: none;"
```
Comment (cleargif) [inline app/views/external_users/claims/misc_fees/advocates/_misc_fee_fields.html.haml:6]: Using the inline style as the jquery will simply update it inline as well

File: app/views/shared/_supplier_number_fields.html.haml [hunk 5]
```diff
@@ -4,13 +4,13 @@
     %span.fx-numberedList-number
     - editable = f.object.new_record? || f.object.supplier_number.blank? || !f.object.valid?
     - if editable
-      = link_to_remove_association t('.remove'), f, { wrapper_class: 'supplier-number-group', class: 'header-action' }
+      = link_to_remove_association t('.remove'), f, wrapper_class: 'supplier-number-group', class: "header-action fx-numberedList-action", style: "display: none;"
```
Comment (cleargif) [inline app/views/shared/_supplier_number_fields.html.haml:7]: As before

File: app/views/shared/_supplier_number_fields.html.haml [hunk 6]
```diff
@@ -4,13 +4,13 @@
...
     - else
       - if f.object.has_non_archived_claims?
         %span.form-hint
           Cannot remove supplier number while claims are still in progress
       - else
-        = link_to_remove_association t('.remove'), f, { wrapper_class: 'supplier-number-group', class: 'header-action' }
+        = link_to_remove_association t('.remove'), f, wrapper_class: 'supplier-number-group', class: "header-action fx-numberedList-action", style: "display: none;"
```
Comment (cleargif) [inline app/views/shared/_supplier_number_fields.html.haml:13]: As before

**Review by colinbruce — APPROVED** [review]
Looks good.  :+1:

**Pull request merged**